lightkurve/targetpixelfile.py

```diff
@@ -537,3 +543,195 @@ def get_bkg_lightcurve(self, aperture_mask=None):
         aperture_mask = self._parse_aperture_mask(aperture_mask)
         return LightCurve(flux=np.nansum(self.flux_bkg[:, aperture_mask], axis=1),
                           time=self.time, flux_err=self.flux_bkg_err)
+
+
+class TargetPixelFileFactory(object):
+
+    def __init__(self, n_cadences, n_rows, n_cols, target_id="unnamed-target",
+                 keywords={}):
+        self.n_cadences = n_cadences
+        self.n_rows = n_rows
+        self.n_cols = n_cols
+        self.target_id = target_id
+        self.keywords = keywords
+
+        # Initialize the 3D data structures
+        self.raw_cnts = np.empty((n_cadences, n_rows, n_cols), dtype='int')
+        self.flux = np.empty((n_cadences, n_rows, n_cols), dtype='float32')
+        self.flux_err = np.empty((n_cadences, n_rows, n_cols), dtype='float32')
+        self.flux_bkg = np.empty((n_cadences, n_rows, n_cols), dtype='float32')
+        self.flux_bkg_err = np.empty((n_cadences, n_rows, n_cols), dtype='float32')
+        self.cosmic_rays = np.empty((n_cadences, n_rows, n_cols), dtype='float32')
+
+        # Set 3D data defaults
+        self.raw_cnts[:, :, :] = -1
```

Comment:
You could do: ``self.raw_cnts[:] = -1`` to set all elements of ``self.raw_cnts`` to ``-1`` (independently of the number of dimensions).

Reply:
I actually like that this is easy to read/ see that the variable has 3 dimensions.